### PR TITLE
Implement the status_code package

### PR DESCRIPTION
**Pull request by supl**
This PR implements the status_code package that defines Scalar DL ledger status codes.

The implementation is very simple. It is just the enumeration definition.

### CODE REVIEW

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

File: ledger/status_code/status_code.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package status_code
+
+// StatusCode hints what happened after a gRPC request are handled.
```
Comment (supl) [inline ledger/status_code/status_code.go]:
```suggestion
// StatusCode hints at what happened after a gRPC request is handled.
```

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**